x-pack/plugin/relevance-search/src/test/java/org/elasticsearch/xpack/relevancesearch/query/RelevanceMatchQueryIntTests.java

```diff
@@ -26,50 +27,99 @@
-            .setSource(Map.of("intField", 12, "doubleField", 13.45))
-            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
-            .execute()
-            .actionGet();
+    public void testNoTextFieldsWithoutSettings() {
+
+        indexDocument(DOCUMENTS_INDEX, "2", Map.of("intField", 12, "doubleField", 13.45));
 
-        RelevanceMatchQueryBuilder relevanceMatchQueryBuilder = new RelevanceMatchQueryBuilder();
         relevanceMatchQueryBuilder.setQuery("text match");
-        final SearchRequestBuilder searchRequestBuilder = client().prepareSearch("index").setQuery(relevanceMatchQueryBuilder);
+        final SearchRequestBuilder searchRequestBuilder = client().prepareSearch(DOCUMENTS_INDEX).setQuery(relevanceMatchQueryBuilder);
 
         final String expectedMsg = "[relevance_match] query cannot find text fields in the index";
         assertFailures(searchRequestBuilder, RestStatus.BAD_REQUEST, containsString(expectedMsg));
     }
+
+    public void testFieldSettings() {
+        final String settingsId = "test-settings";
+        indexDocument(
+            RelevanceSettingsService.ENT_SEARCH_INDEX,
+            RelevanceSettingsService.RELEVANCE_SETTINGS_PREFIX + settingsId,
+            Map.of("fields", List.of("textField"))
+        );
+
+        indexDocument(
+            DOCUMENTS_INDEX,
+            "1",
+            Map.of("textField", "text example", "intField", 12, "doubleField", 13.45, "anotherTextField", "should match")
+        );
+        indexDocument(
+            DOCUMENTS_INDEX,
+            "2",
+            Map.of("textField", "other document", "intField", 12, "doubleField", 13.45, "anotherTextField", "text example")
+        );
+
+        relevanceMatchQueryBuilder.setQuery("text example");
+        relevanceMatchQueryBuilder.setRelevanceSettingsId(settingsId);
+        SearchResponse response = client().prepareSearch(DOCUMENTS_INDEX).setQuery(relevanceMatchQueryBuilder).get();
+
+        assertHitCount(response, 1);
+        assertSearchHits(response, "1");
+    }
+
+    public void testFieldSettingsNotFound() {
+        relevanceMatchQueryBuilder.setQuery("text example");
+        final String relevanceSettingsId = "non-existing-settings";
+        relevanceMatchQueryBuilder.setRelevanceSettingsId(relevanceSettingsId);
+
+        final SearchRequestBuilder searchRequestBuilder = client().prepareSearch(DOCUMENTS_INDEX).setQuery(relevanceMatchQueryBuilder);
+
+        final String expectedMsg = "[relevance_match] query can't find search settings: " + relevanceSettingsId;
+        assertFailures(searchRequestBuilder, RestStatus.BAD_REQUEST, containsString(expectedMsg));
+    }
+
+    private void indexDocument(String index, String id, Map<String, Object> document) {
+        client().prepareIndex(index)
+            .setId(id)
+            .setSource(document)
+            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+            .execute()
+            .actionGet();
```

Comment:
I'm a dummy here, but why `actionGet`? If we index a document, wouldn't this be a `post/put`?

Reply:
`actionGet` is pretty confusing; this does not refer to the actual GET, but to the `ActionFuture` interface that `execute()` returns. `actionGet` is equivalent to `Future.get()` but transforming some exceptions.